prefect_gcp/cloud_storage.py

```diff
@@ -1140,3 +1141,77 @@ async def upload_from_folder(
         if num_uploaded == 0:
             self.logger.warning(f"No files were uploaded from {from_folder}.")
         return bucket_folder
+
+    @sync_compatible
+    async def upload_from_dataframe(
+        self,
+        df: pd.DataFrame,
+        to_path: str,
+        output_format: str = "parquet",
+        compression: Optional[str] = None,
+        **upload_kwargs: Dict[str, Any],
+    ) -> str:
+        """
+        Upload a pandas Dataframe to Google Cloud Storage as:
+            .csv, .csv.gz, .parquet, .parquet.snappy, .parquet.gz
+
+        Args:
+            df (pd.DataFrame): pandas Dataframe object
+            to_path (str): the actual full blob name (e.g.: /path/to/gcs/blob.csv)
+            output_format (str): Specify whether the output should be csv or parquet
+            compression (Optional[str], optional):
+                Specify the compression type for the output.
+                'csv' supports 'None' or 'gzip',
+                'parquet' supports 'None', 'snappy', 'gzip'.
+                Defaults to None.
+            **upload_kwargs: Additional keyword arguments to pass to
+                `Blob.upload_from_file`.
+
+        Returns:
+            The path that the folder was uploaded to.
+
+        Examples:
+            Upload local folder my_folder to the bucket's folder my_folder.
+            ```python
+            from prefect_gcp.cloud_storage import GcsBucket
+
+            gcs_bucket = GcsBucket.load("my-bucket")
+            gcs_bucket.upload_from_dataframe(
+                df=pandas_df, blob_name="/path/to/gcs/blob.parquet",
+                output='parquet', compression='snappy'
+            )
+            ```
+        """
+        byte_buffer = BytesIO()
```

Comment:
I think we might want to move this down a bit and use it as a context manager:

with BytesIO() as bytes_buffer
    ...

Reply:
Fixed 👍🏻